ntoskrnl/mm/balance.c

```diff
@@ -372,6 +379,14 @@ MiBalancerThread(PVOID Unused)
                     InitialTarget = MiTrimMemoryConsumer(i, InitialTarget);
                 }
 
+                /* Trim cache */
+                Target = max(InitialTarget, MiMinimumAvailablePages - MmAvailablePages);
```

Comment:
This line results in a request for a HUGE amount of pages to be freed as can be seem from @JoachimHenze's comment including his debug output above showing the requested pages:
```
(/ntoskrnl/cc/view.c:473) CcRosTrimCache(Target >>4294848615<<)
```
This subtract for this one case is in the wrong order. I suggest this change to fix all cases:
```
Target = max(InitialTarget, abs(MiMinimumAvailablePages - MmAvailablePages));
```

Reply:
It may actually be worth studying both cases:
1. MiMinimumAvailablePages < MmAvailablePages
2. and MmAvailablePages < MiMinimumAvailablePages , which is what seems to happen.

Note that elsewhere in that file we can see this:
https://git.reactos.org/?p=reactos.git;a=blob;f=ntoskrnl/mm/balance.c;hb=8532f1874a34ae41818a770f8c8bd8a77b70e21a#l111
```c
 111     if (MmAvailablePages < MiMinimumAvailablePages)
 112     {
 113         /* Global page limit exceeded */
 114         Target = (ULONG)max(Target, MiMinimumAvailablePages - MmAvailablePages);
 115     }
 116     else if (MiMemoryConsumers[Consumer].PagesUsed > MiMemoryConsumers[Consumer].PagesTarget)
 117     {
 118         /* Consumer page limit exceeded */
 119         Target = max(Target, MiMemoryConsumers[Consumer].PagesUsed - MiMemoryConsumers[Consumer].PagesTarget);
 120     }
```

Reply:
I agree that changing the code at this location to match what you show from above is a good solution here. Thanks.